Contents/Code/siteUltrafilmsxxx.py

```diff
@@ -0,0 +1,108 @@
+import PAsearchSites
+import PAgenres
+import PAactors
+import PAextras
+import ssl
+from lxml.html.soupparser import fromstring
+
+# maybe helpful for linux users, who has "sslv3 alert handshake failure (_ssl.c:590)>" @kamuk90
+def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
+    if searchSiteID != 9999:
+        siteNum = searchSiteID
+    url = PAsearchSites.getSearchSearchURL(siteNum) + encodedTitle
+    headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'}
+
+    try:
+        searchResults = HTML.ElementFromURL(url)
+    except:
+        request = urllib.Request(url, headers=headers)
+        response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
+        htmlstring = response.read()
+        searchResults = fromstring(htmlstring)
+
+    for searchResult in searchResults.xpath('//ul[@class="listing-videos listing-tube"]'):
+        titleNoFormatting = searchResult.xpath('li[@class="border-radius-5 box-shadow"]//a')[0].get('title')
+        Log("titleNoFormatting: " + titleNoFormatting)
+        actors = searchResults.xpath('//li[@class="border-radius-5 box-shadow"]//img')[0].get('alt').replace(' - ', '').replace(titleNoFormatting, '')
+        sceneUrl = searchResult.xpath('//li[@class="border-radius-5 box-shadow"]//a')[0].get('href')
+        curID = sceneUrl.replace('/','_').replace('?','!')
+        Log("curID: " + curID)
+        score = 100
+        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " with "+ actors + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] ", score = score, lang = lang))
+    return results
+
+def update(metadata,siteID,movieGenres,movieActors):
+    pageURL = str(metadata.id).split("|")[0].replace('_', '/')
+    Log('scene url: ' + pageURL)
+    headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'}
+    try:
+        detailsPageElements = HTML.ElementFromURL(pageURL)
+    except:
+        request = urllib.Request(pageURL, headers=headers)
+        response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
+        htmlstring = response.read()
+        detailsPageElements = fromstring(htmlstring)
+
+    # Summary
+    siteName = PAsearchSites.getSearchSiteName(siteID)
+    metadata.studio = siteName
+    try:
+        #site dont give a description
+        metadata.summary = 'No description'
+        metadata.title = detailsPageElements.xpath('//span[@property="name"]')[3].text_content().strip()
+    except:
+        pass
+
+    # Collections / Tagline
+    metadata.collections.clear()
+    metadata.tagline = siteName
+    metadata.collections.add(siteName)
+
+    # Genres
+    try:
+        genres = detailsPageElements.xpath('//div[@itemprop="keywords"]//ul//li')
```

Comment:
and this

genres = detailsPageElements.xpath('//div[@itemprop="keywords"]//ul//li//a')